portal-ui/src/screens/Console/Tenants/TenantDetails/TenantYAML.tsx

```diff
@@ -0,0 +1,180 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { useEffect, useState } from "react";
+import { connect } from "react-redux";
+import Grid from "@material-ui/core/Grid";
+import { Button, LinearProgress } from "@material-ui/core";
+import { createStyles, Theme, withStyles } from "@material-ui/core/styles";
+import api from "../../../../common/api";
+import {
+  setErrorSnackMessage,
+  setModalErrorSnackMessage,
+} from "../../../../actions";
+import {
+  fieldBasic,
+  modalBasic,
+} from "../../Common/FormComponents/common/styleLibrary";
+import ModalWrapper from "../../Common/ModalWrapper/ModalWrapper";
+import InputBoxWrapper from "../../Common/FormComponents/InputBoxWrapper/InputBoxWrapper";
+import CodeMirrorWrapper from "../../Common/FormComponents/CodeMirrorWrapper/CodeMirrorWrapper";
+import { HasPermissionResponse } from "../../Buckets/types";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    jsonPolicyEditor: {
+      minHeight: 400,
+      width: "100%",
+    },
+    buttonContainer: {
+      textAlign: "right",
+    },
+    ...modalBasic,
+    ...fieldBasic,
+  });
+
+interface ITenantYAML {
+  yaml: string;
+}
+
+interface ITenantYAMLProps {
+  classes: any;
+  open: boolean;
+  closeModalAndRefresh: (refresh: boolean) => void;
+  tenant: string;
+  namespace: string;
+  setModalErrorSnackMessage: typeof setModalErrorSnackMessage;
+}
+
+const TenantYAML = ({
+  classes,
+  open,
+  closeModalAndRefresh,
+  tenant,
+  namespace,
+}: ITenantYAMLProps) => {
+  const [addLoading, setAddLoading] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [tenantYaml, setTenantYaml] = useState<string>("");
+
+  const updateTenant = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (addLoading) {
+      return;
+    }
+    setAddLoading(true);
+    api
+      .invoke("PUT", `/api/v1/namespaces/${namespace}/tenants/${tenant}/yaml`, {
+        yaml: tenantYaml,
+      })
+      .then((res) => {
+        setAddLoading(false);
+        closeModalAndRefresh(true);
+      })
+      .catch((err) => {
+        setAddLoading(false);
+        setModalErrorSnackMessage(err);
+      });
+  };
+
+  // check the permissions for creating bucket
+  useEffect(() => {
+    api
+      .invoke("GET", `/api/v1/namespaces/${namespace}/tenants/${tenant}/yaml`)
+      .then((res: ITenantYAML) => {
+        setLoading(false);
+        setTenantYaml(res.yaml);
+      })
+      .catch((err: any) => {
+        setLoading(false);
+        setErrorSnackMessage(err);
+      });
+  }, []);
```

Comment:
Please add namespace & tenant vars into the dependency array
```suggestion
  }, [namespace, tenant]);
```

Reply:
done